PyRDF/Node.py

```diff
@@ -22,27 +28,46 @@ def __init__(self, _get_head, operation):
         self.value = None
 
     def __getattr__(self, attr):
-        self._cur_attr = attr
+        # Interception of operation calls
+
+        self._cur_attr = attr # Store new operation name
         return self._call_handler
 
     def _call_handler(self, *args, **kwargs):
 
+        # Create a new `Operation` object for the
+        # incoming operation call
         op = Operation(self._cur_attr, *args, **kwargs)
+
+        # Create a new `Node` object to house the operation
         newNode = Node(operation=op, _get_head=self._get_head)
+
+        # Add the new node as a child of the current node
         self.next_nodes.append(newNode)
 
+        # Return a Proxy object if the new node
+        # is an action node
         if op.op_type == Operation.Types.ACTION:
             return Proxy(newNode)
 
         return newNode
     
     def graph_prune(self):
+        """
+        A Function to prune the nodes
+        of a PyRDF graph based on
+        user references
+
```

Comment:
Elaborate a bit more: A node will be pruned if it has no children and the user application does not hold any reference to it.